### PR TITLE
[kubelet] Use a generic Set instead of a specified Set

### PR DESCRIPTION
**Pull request by bzsuni**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Specified Set is deprecated, we need use generic Set instead.
According to https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go#L25
```
// String is a set of strings, implemented via map[string]struct{} for minimal memory consumption.
//
// Deprecated: use generic Set instead.
// new ways:
// s1 := Set[string]{}
// s2 := New[string]()
type String map[string]Empty
```
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by bzsuni** [comment]
/assign @msau42  @sjenning
For approval, thanks

File: test/e2e/framework/ssh/ssh.go [hunk 1]
```diff
@@ -86,6 +86,11 @@ func GetSigner(provider string) (ssh.Signer, error) {
 		if keyfile == "" {
 			keyfile = "id_rsa"
 		}
+	case "azure":
```
Comment (bart0sh) [inline test/e2e/framework/ssh/ssh.go:89]: @bzsuni this looks unrelated to the rest of PR changes. Would it make sense to submit it separately?

Comment (bzsuni) [reply test/e2e/framework/ssh/ssh.go:89]: There is a problem with my commit, I will make some changes later

Comment (bzsuni) [reply test/e2e/framework/ssh/ssh.go]: I have made the necessary modifications, thanks @bart0sh

**Comment by bzsuni** [comment]
/assign @yujuhong 
And this PR also need your review please

**Comment by bzsuni** [comment]
/assign @liggitt 
for approval please
